pkg/v7/configmap/desired.go

```diff
@@ -278,6 +298,29 @@ func newConfigMapLabels(configMapValues ConfigMapValues, appName, projectName st
 	}
 }
 
+func newConfigMapSpecs(providerChartSpecs []key.ChartSpec) []ConfigMapSpec {
```

Comment:
newConfigMapSpecs generates the spec structs from the key functions. Provider chart specs will be injected from the provider resources.